crates/rpc-types-eth/src/transaction/request.rs

```diff
@@ -178,104 +178,125 @@ impl TransactionRequest {
+                .ok_or("Missing 'max_priority_fee_per_gas' field for Eip1559 transaction.")?,
+            max_fee_per_gas: self
+                .max_fee_per_gas
+                .ok_or("Missing 'max_fee_per_gas' field for Eip1559 transaction.")?,
+            gas_limit: self.gas.ok_or("Missing 'gas_limit' field for Eip1559 transaction.")?,
             to: checked_to,
             value: self.value.unwrap_or_default(),
             input: self.input.into_input().unwrap_or_default(),
             access_list: self.access_list.unwrap_or_default(),
-        }
+        })
     }
 
     /// Build an EIP-2930 transaction.
     ///
-    /// # Panics
-    ///
-    /// If required fields are missing. Use `complete_2930` to check if the
+    /// Returns an error if required fields are missing. Use `complete_2930` to check if the
     /// request can be built.
-    fn build_2930(self) -> TxEip2930 {
-        let checked_to = self.to.expect("checked in complete_2930.");
+    fn build_2930(self) -> Result<TxEip2930, &'static str> {
+        let checked_to = self.to.ok_or("Missing 'to' field for Eip2930 transaction.")?;
 
-        TxEip2930 {
+        Ok(TxEip2930 {
             chain_id: self.chain_id.unwrap_or(1),
-            nonce: self.nonce.expect("checked in complete_2930"),
-            gas_price: self.gas_price.expect("checked in complete_2930"),
-            gas_limit: self.gas.expect("checked in complete_2930"),
+            nonce: self.nonce.ok_or("Missing 'nonce' field for Eip2930 transaction.")?,
+            gas_price: self
+                .gas_price
+                .ok_or("Missing 'gas_price' field for Eip2930 transaction.")?,
+            gas_limit: self.gas.ok_or("Missing 'gas_limit' field for Eip2930 transaction.")?,
             to: checked_to,
             value: self.value.unwrap_or_default(),
             input: self.input.into_input().unwrap_or_default(),
             access_list: self.access_list.unwrap_or_default(),
-        }
+        })
     }
 
-    /// Build an EIP-4844 transaction.
+    /// Build an EIP-4844 transaction variant - either with or without sidecar.
     ///
-    /// # Panics
+    /// Returns an error if required fields are missing. Use `complete_4844` to check if the
+    /// request can be built.
+    fn build_4844_variant(self) -> Result<TxEip4844Variant, &'static str> {
+        if self.sidecar.is_none() {
+            self.build_4844_without_sidecar().map(Into::into)
+        } else {
+            self.build_4844_with_sidecar().map(Into::into)
+        }
+    }
+
+    /// Build an EIP-4844 transaction without sidecar.
     ///
-    /// If required fields are missing. Use `complete_4844` to check if the
+    /// Returns an error if required fields are missing. Use `complete_4844` to check if the
     /// request can be built.
-    fn build_4844(mut self) -> TxEip4844WithSidecar {
-        self.populate_blob_hashes();
+    fn build_4844_without_sidecar(self) -> Result<TxEip4844, &'static str> {
```

Comment:
this is the new function we need, everything elese is unrelated, should not be part of this pr